pkg/utils/status.go

```diff
@@ -654,77 +627,245 @@ func WaitForNginxController(ctx context.Context, instance csmv1.ContainerStorage
 	return wait.PollImmediate(time.Second, timeout, GetNginxControllerStatus(ctx, instance, r))
 }
 
-// checkForServices - Calculate success state for services deployed with app-mob
-func checkForServices(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, newStatus *csmv1.ContainerStorageModuleStatus, isDaemonset bool) (bool, error) {
-
+// statusForAppMob - calculate success state for application-mobility module
+func appMobStatusCheck(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, _ *csmv1.ContainerStorageModuleStatus) (bool, error) {
 	log := logger.GetLogger(ctx)
-	running := false
-	var err error = nil
-	if isDaemonset {
-		expected, nodeStatus, daemonSetErr := getDaemonSetStatus(ctx, instance, r)
-		newStatus.NodeStatus = nodeStatus
-		newStatus.State = constants.Failed
-		log.Infof("daemonset expected [%d]", expected)
-		log.Infof("daemonset nodeStatus.Available [%s]", nodeStatus.Available)
-		if fmt.Sprintf("%d", expected) == nodeStatus.Available {
-			running = true
-			newStatus.State = constants.Succeeded
+	veleroEnabled := false
+	certEnabled := false
+	var certManagerRunning bool
+	var certManagerCainInjectorRunning bool
+	var certManagerWebhookRunning bool
+	appMobRunning := false
+	veleroRunning := false
+	var daemonRunning bool
+	readyPods := 0
+	expected := 2
```

Comment:
Have we tried this on a k8s cluster with worker nodes not equal to 2?
I don't think we should hard code 2 as the expected number here

Reply:
Thanks for the comments 
i have fixed it and verified it